src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2645,5 +2656,71 @@ private void InterpretCall(int token)
             if (!signature.ReturnType.IsVoid)
                 _stack.Push(callInfo.ReturnValue);
         }
+        
+        private unsafe void InterpretNewObj(int token)
+        {
+            MethodDesc method = (MethodDesc)_methodIL.GetObject(token);
+            MethodSignature signature = method.Signature;
+            TypeDesc owningType = method.OwningType;
+
+            StackItem[] arguments = new StackItem[signature.Length];
+            for (int i = 0; i < arguments.Length; i++)
+            {
+                arguments[i] = PopWithValidation();
+            }
+
+            if (owningType.IsArray)
+            {
+                int[] lArguments = new int[arguments.Length];
+                for (int i = arguments.Length - 1; i >= 0; i--)
+                {
+                    lArguments[i] = arguments[i].AsInt32();
+                }
+
+                Array.Reverse(lArguments);
```

Comment:
Fill the array in the right order directly to make this reverse call unnecessary?

Reply:
All Done!